src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs

```diff
@@ -5,6 +5,11 @@
 using Microsoft.CodeAnalysis.Options;
 using Roslyn.Utilities;
 
+#if CODE_STYLE
+using CSharpWorkspaceResources = Microsoft.CodeAnalysis.CSharp.CSharpCodeStyleResources;
+using WorkspacesResources = Microsoft.CodeAnalysis.CodeStyleResources;
+#endif
+
 namespace Microsoft.CodeAnalysis.CSharp.Formatting
 {
     public static partial class CSharpFormattingOptions
```

Comment:
>CSharpFormattingOptions [](start = 32, length = 23)

So will this type be exposed as a public API from both CodeStyle layer and Workspaces layer? I don't see how this will go away with any compiler changes..

Reply:
Regardless, you don't need to be concerned with this now. Probably, I am not seeing the big picture of your future refactoring when the compiler's AnalyzerConfigOptions is indeed available. I will look at the public API file for CodeStyle layer after that refactoring to understand this better.

---
In reply to: [236814228](https://github.com/dotnet/roslyn/pull/31276#discussion_r236814228) [](ancestors = 236814228)

Reply:
> So will this type be exposed as a public API from both Code Style layer and Workspaces layer? 

The CodeStyle layer does not currently represent an API surface area.

> I don't see how this will go away with any compiler changes.

The formatter will be rewritten to use a lower-level options representation, eliminating the need to reference `IOption` and related types in the code style layer.